index.js

```diff
@@ -6,10 +6,14 @@ module.exports = function (options) {
     options = assign({
         el : undefined,
         parent: document,
-        filter: undefined
+        filter: undefined,
+        matchParent: false
     }, options);
     curNode = options.el;
-    while(
+    if (options.matchParent && curNode === options.parent) {
+        return curNode;
+    }
```

Comment:
Shouldn't this happen after the while loop? What if you want to match the parent, but `curNode` starts as a descendent of `options.parent`?